assignment2/index.js

```diff
@@ -0,0 +1,103 @@
+var express = require("express");
+var mongoose = require("mongoose");
+var passwordHash = require("password-hash");
+var app = express();
+var bodyParser = require("body-parser");
+var async = require("async");
+const bcrypt = require("bcrypt");
+var cookieParser = require("cookie-parser");
+var { auth } = require("./middleware/auth");
+var { conn, user } = require("./config");
+app.use(cookieParser());
+const saltRounds = 10;
+var urlencodedParser = bodyParser.urlencoded({ extended: false });
+app.use(express.json());
+
+app.post("/user/register", async function (req, res) {
+  console.log(req.body.username);
+  const salt = await bcrypt.genSalt();
+  const userPassword = await bcrypt.hash(req.body.password, salt);
+  const user_post = new user({
+    username: req.body.username,
+    password: userPassword,
+    email: req.body.email,
+  });
```

Comment:
do this after checking of the password and confirmation password are same.

Reply:
done sir